Doc/whatsnew/3.7.rst

```diff
@@ -483,138 +782,458 @@ The :func:`~binascii.b2a_uu` function now accepts an optional *backtick*
+(Jelle Zijlstra with further improvements by INADA Naoki, Serhiy Storchaka,
+and Raymond Hettinger in :issue:`28638`.)
+
+Additionally, ``collections.namedtuple()`` now supports default values.
+(Contributed by Raymond Hettinger in :issue:`32320`.)
+
+:class:`collections.ChainMap` now preserves the order of the underlying
+mappings.  (Contributed by Raymond Hettinger in :issue:`32792`.)
+
+
+compileall
+----------
+
+:func:`compileall.compile_dir` learned the new *invalidation_mode* parameter,
+which can be used to enable
+:ref:`hash-based .pyc invalidation <whatsnew37-pep552>`.  The invalidation
+mode can also be specified on the command line using the new
+``--invalidation-mode`` argument.
+(Contributed by Benjamin Peterson in :issue:`31650`.)
+
+
+concurrent.futures
+------------------
+
+:class:`ProcessPoolExecutor <concurrent.futures.ProcessPoolExecutor>` and
+:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>` now
+support the new *initializer* and *initargs* constructor arguments.
+(Contributed by Antoine Pitrou in :issue:`21423`.)
+
+The :class:`ProcessPoolExecutor <concurrent.futures.ProcessPoolExecutor>`
+can now take the multiprocessing context via the new *mp_context* argument.
+(Contributed by Thomas Moreau in :issue:`31540`.)
+
+``executor.submit()`` now raises a :exc:`RuntimeError` if called during
+interpreter shutdown.
+(Contributed by Mark Nemec in :issue:`33097`.)
+
+
+configparser
+------------
+
+The :class:`~configparser.ConfigParser`` constructor now uses ``read_dict()``
+to process the default values, making its behavior consistent with the
+rest of the parser.  Non-string keys and values in the defaults
+dictionary are now being implicitly converted to strings.
+(Contributed by James Tocknell. in :issue:`23835`.)
+
+The *filename* parameter of the
+:meth:`ConfigParser.read() <configparser.ConfigParser.read>` method now
+accepts bytes objects.
+(Contributed by Vincent Michel. in :issue:`31307`.)
+
+
 contextlib
 ----------
 
-:func:`~contextlib.asynccontextmanager` and
-:class:`~contextlib.AbstractAsyncContextManager` have been added. (Contributed
+The new :func:`~contextlib.nullcontext` is a simpler and faster alternative
+to :class:`~contextlib.ExitStack` for handling simple context managers.
+(Contributed by Jesse-Bakker in :issue:`10049`.)
+
+The new :func:`~contextlib.asynccontextmanager` and
+:class:`~contextlib.AbstractAsyncContextManager` have been added to
+complement their synchronous counterparts.  (Contributed
 by Jelle Zijlstra in :issue:`29679` and :issue:`30241`.)
 
-:class:`contextlib.AsyncExitStack` has been added. (Contributed by
+The new :class:`contextlib.AsyncExitStack` class has been added as an
+asynchronous equivalent to :class:`contextlib.ExitStack`.  (Contributed by
 Alexander Mohr and Ilya Kulakov in :issue:`29302`.)
 
+
 cProfile
 --------
 
 :mod:`cProfile` command line now accepts ``-m module_name`` as an alternative
```

Comment:
Nit: for consistency with other entries, this sentence should start with "The ".